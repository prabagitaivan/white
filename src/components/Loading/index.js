import React, { memo } from 'react'
import { LinearProgress } from '@material-ui/core'
import { makeStyles } from '@material-ui/styles'

const useStyles = makeStyles(
  {
    root: {
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
      marginTop: '40vh'
    },
    linearProgress: {
      height: 10,
      width: '60%',

      '& .MuiLinearProgress-barColorSecondary': {
        backgroundColor: '#b1b1b1'
      }
    }
  },
  { name: 'Loading' }
)

export default memo(() => {
  const classes = useStyles()

  return (
    <div className={classes.root}>
      <div>Loading</div>
      <LinearProgress color='secondary' className={classes.linearProgress} />
    </div>
  )
})
