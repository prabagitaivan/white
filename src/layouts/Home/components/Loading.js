import React, { PureComponent } from 'react'
import { LinearProgress } from '@material-ui/core'
import { loading as styles } from '../styles'

export default class Loading extends PureComponent {
  render () {
    return (
      <div style={styles.root}>
        <div>Loading</div>
        <LinearProgress color='secondary' style={styles.linearProgress} />
      </div>
    )
  }
}
