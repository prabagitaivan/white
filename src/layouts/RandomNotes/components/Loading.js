import React, { memo } from 'react'
import { LinearProgress } from '@material-ui/core'
import { loading as useStyles } from '../styles'

export default memo(() => {
  const classes = useStyles()

  return (
    <div className={classes.root}>
      <div>Loading</div>
      <LinearProgress color='secondary' className={classes.linearProgress} />
    </div>
  )
})
