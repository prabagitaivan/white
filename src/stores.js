import { createStore as ReduxCreateStore, applyMiddleware } from 'redux'
import createSagaMiddleware from 'redux-saga'
import reducers from './reducers'
import sagas from './sagas'

export const createStore = preloadedState => {
  const sagaMiddleware = createSagaMiddleware()
  const enhancer = applyMiddleware(sagaMiddleware)

  const store = ReduxCreateStore(reducers, preloadedState, enhancer)
  sagas.forEach(sagaMiddleware.run)

  return store
}

export default createStore()
