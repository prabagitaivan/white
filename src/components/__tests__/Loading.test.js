import { render } from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import { createStore } from '../../stores'
import Styles from '../../styles'
import Loading from '../Loading'

describe('components Loading', () => {
  describe('snapshots', () => {
    it('return correct snapshot', () => {
      const { container } = render(
        <Provider store={createStore()}>
          <Styles>
            <Loading />
          </Styles>
        </Provider>
      )
      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
