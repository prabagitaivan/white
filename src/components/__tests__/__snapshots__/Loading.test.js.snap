// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Loading snapshots return correct snapshot 1`] = `
<div
  class="WhiteLoading-root"
>
  <div>
    Loading
  </div>
  <span
    class="MuiLinearProgress-root MuiLinearProgress-colorSecondary MuiLinearProgress-indeterminate WhiteLoading-linearProgress css-m48q12-MuiLinearProgress-root"
    role="progressbar"
  >
    <span
      class="MuiLinearProgress-bar MuiLinearProgress-barColorSecondary MuiLinearProgress-bar1Indeterminate css-3v28zu-MuiLinearProgress-bar1"
    />
    <span
      class="MuiLinearProgress-bar MuiLinearProgress-barColorSecondary MuiLinearProgress-bar2Indeterminate css-vnqxk6-MuiLinearProgress-bar2"
    />
  </span>
</div>
`;
